### PR TITLE
OSSM-5848 Added feature annotations

### PR DESCRIPTION
**Pull request by DeanDoyle1502**
Added feature annotations

- Wondering if the values are correct

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -32,6 +32,16 @@ csv:
...
+  olmFeatureDisconnected: true
+  olmFeatureFIPSCompliant: true
+  olmFeatureProxyAware: true
+  olmFeatureTLSProfiles: false
+  olmFeatureTokenAuthAWS: false
+  olmFeatureTokenAuthAzure: false
+  olmFeatureTokenAuthGCP: false
+  olmFeatureCNF: false
+  olmFeatureCNI: true
+  olmFeatureCSI: false
```
Comment (luksa) [inline chart/values.yaml]: We should probably group these under a common parent (e.g. `olmFeatures`) and remove the `olmFeature` prefix.

File: bundle/manifests/sailoperator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -23,8 +23,18 @@ metadata:
...
     categories: OpenShift Optional, Integration & Delivery, Networking, Security
     containerImage: quay.io/maistra-dev/istio-operator:3.0-latest
-    createdAt: "2024-02-08T11:24:58Z"
+    createdAt: "2024-02-12T10:02:23Z"
     description: Experimental operator for installing Istio service mesh
+    features.operators.openshift.io/cnf: "false"
+    features.operators.openshift.io/cni: "true"
+    features.operators.openshift.io/csi: "false"
+    features.operators.openshift.io/disconnected: "true"
+    features.operators.openshift.io/fips-compliant: "true"
```
Comment (luksa) [inline bundle/manifests/sailoperator.clusterserviceversion.yaml]: This should probably be false, at least for now (Sail doesn't yet use our build of the proxy). Also note that the Sail operator is the equivalent of the community operator in 2.x. I believe we set the values for the product build outside of this repository. @FilipB should know more about this.

Comment (DeanDoyle1502) [reply bundle/manifests/sailoperator.clusterserviceversion.yaml]: Thanks Marko changes made there

File: chart/values.yaml [hunk 3]
```diff
@@ -32,6 +32,18 @@ csv:
...
+    disconnected: false
+    FIPSCompliant: false
+    proxyAware: false
+    TLSProfiles: false
+    tokenAuthAWS: false
+    tokenAuthAzure: false
+    tokenAuthGCP: false
+    CNF: false
+    CNI: false
+    CSI: false
```
Comment (luksa) [inline chart/values.yaml]: @dgn do we want these to start with a lowercase character?

File: bundle/manifests/sailoperator.clusterserviceversion.yaml (hunk 2 above)
Comment (FilipB) [reply bundle/manifests/sailoperator.clusterserviceversion.yaml]: Better to check with OLM team, but I also think we should not include anything OCP specific to the community operator. On the other hand we will want to run community operator on both OCP and K8s so question for OLM team is how to do that.

**Comment by dgn** [comment]
I would prefer to have these in values.yaml with their full names and values through a generic `csv.annotations` field

**Comment by DeanDoyle1502** [comment]
-@dgn Added new commit there with what I think you were looking for , if its not i can change again.

-Still having trouble getting answer for values

File: chart/values.yaml [hunk 4]
```diff
@@ -32,6 +32,17 @@ csv:
   - istio
   - servicemesh
   - envoy
+  annotations: 
+    olmFeatureDisconnected: false
```
Comment (dgn) [inline chart/values.yaml]: what I meant is to have the actual annotations in here (e.g. `features.operators.openshift.io/disconnected: true`) and in the template just looping over the annotation key-value pairs and adding them

Comment (dgn) [reply chart/values.yaml]: similar to how env vars are added in this upstream chart:
https://github.com/istio/istio/blob/ef08bf9850209ed8ecac3065a92b40427d9425f0/manifests/charts/istio-control/istio-discovery/templates/deployment.yaml#L151-L156

**Comment by dgn** [comment]
For the actual values, we can set `disconnected` and `cni` to `true` for now and leave everything else at `false`

File: chart/values.yaml (hunk 4 above)
Comment (DeanDoyle1502) [reply chart/values.yaml]: Made those changes as per what i think your looking for, let me know if there's anything not correct 

-changed values of disconnected and cni to true also

**Review by luksa — APPROVED** [review]

**Review by dgn — APPROVED** [review]
Thank you @DeanDoyle1502!

**Review by FilipB — APPROVED** [review]
We have to remember to set fips to true when we are done with fips testing.

**Pull request merged**